src/java/org/apache/cassandra/db/compaction/unified/StaticController.java

```diff
@@ -168,6 +168,11 @@ public static Map<String, String> validateOptions(Map<String, String> options) t
         return options;
     }
 
+    protected void setScalingParameters(String newScalingParameters)
+    {
+        scalingParameters = parseScalingParameters(newScalingParameters);
```

Comment:
this doesn't look thread-safe with `getScalingParameter`.

For example, when executing `getScalingParameter`:
1. `index` = 5 and `scalingParameters.length`  is 6
2. `scalingParameters` is updated via this API and became size of 3.
3. `scalingParameters[index=5]` will throw index out of bound

I think we need to:
1. make `scalingParameters` volatile
2. either synchronize `setScalingParameters` and `getScalingParameter` or copy existing `scalingParameters` to a local variable `currentScalingParameters` and use the coped variable in `getScalingParameter`